ivy/functional/frontends/paddle/tensor/creation.py

```diff
@@ -66,3 +66,13 @@ def arange(start, end=None, step=1, dtype=None, name=None):
 @to_ivy_arrays_and_back
 def empty(shape, dtype=None):
     return ivy.empty(shape=shape, dtype=dtype)
+
+
+@to_ivy_arrays_and_back
+def diag(x, offset=0, padding_value=0, name=None):
+    d = ivy.diag(x, k=offset)
+    if len(x.shape) == 1:
+        if padding_value:
+            d_mask = ivy.eye(d.shape[0], k=offset)
+            d[d_mask] = padding_value
+    return d

```

Comment:
Try implementing this using `ivy.diagflat` instead. It handles the padding argument for you